docs/source/torch.compiler_deepdive.rst

```diff
@@ -210,53 +210,128 @@ point for the first time.
+
+The graph break is shown in the code of compiled ``toy_example``, where we have to use python interpreter to select the following graph to execute.
+
+Note that we pass a simple ``my_compiler`` function as the backend compiler, therefore the subgraph code ``__resume_at_38_2``, ``__resume_at_30_1``, and ``__compiled_fn_0`` remain python code. This can also be inspected (please ignore the function name, and only use the function signature and function body code):
+
+.. code-block:: python
 
-   def __resume_at_30_1(b, x):
+   print("source code of __compiled_fn_0:")
+   print(__compiled_fn_0._torchdynamo_orig_callable.__self__)
+   print("=" * 60)
+   print("source code of __resume_at_30_1:")
+   print(decompile(__resume_at_30_1))
+   print("=" * 60)
+   print("source code of __resume_at_38_2:")
+   print(decompile(__resume_at_38_2))
+
+::
+
+   source code of __compiled_fn_0:
+   GraphModule()
+
+
+
+   def forward(self, L_a_ : torch.Tensor, L_b_ : torch.Tensor):
+       l_a_ = L_a_
+       l_b_ = L_b_
+       abs_1 = torch.abs(l_a_)
+       add = abs_1 + 1;  abs_1 = None
+       truediv = l_a_ / add;  l_a_ = add = None
+       sum_1 = l_b_.sum();  l_b_ = None
+       lt = sum_1 < 0;  sum_1 = None
+       return (truediv, lt)
+
+   # To see more debug info, please use `graph_module.print_readable()`
+   ============================================================
+   source code of __resume_at_30_1:
+   def <resume in toy_example>(b, x):
        b = b * -1
        return x * b
 
-   def fn(a, b):
-       x = a / (torch.abs(a) + 1)
-       lt = b.sum() < 0
-       return x, lt
+   ============================================================
+   source code of __resume_at_38_2:
+   def <resume in toy_example>(b, x):
+       return x * b
+
+However, keep in mind that, if we use other backends like the built-in `inductor`, the subgraph code will be compiled CUDA kernels for GPU or C++ code for CPU.
+
+To summarize, the compiled code is conceptually equivalent to the code below:
+
+.. code-block:: python
+
+   def compiled_example(a, b):
+       L = {'a': a, 'b': b}
+       for guard, code in get_cache_entries():
+           if guard(L):
+               return code(a, b)
+       recompile_and_add_another_cache_entry()
 
-   __compiled_fn_0._torchdynamo_orig_callable = fn
+The following diagram demonstrates how ``torch.compile`` transforms and optimizes user-written code:
 
-Note that we pass a simple `my_compiler` function as the backend compiler, therefore the subgraph code `__resume_at_38_2`, `__resume_at_30_1`, and `__compiled_fn_0._torchdynamo_orig_callable` remain python code. However, if we use other backends like the built-in `inductor`, the subgraph code will be compiled CUDA kernels for GPU or C++ code for CPU.
+.. image:: _static/img/dynamo/flowchart.jpg
```

Comment:
Is it possible to add a paragraph that describes what is going on on the diagram?

Reply:
Sure, fixed in [454676e](https://github.com/pytorch/pytorch/pull/108147/commits/454676e32916ce8381e10a111cf98d589cb879ac).